### PR TITLE
docs: add CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by jtieri**
This PR is a first pass at establishing a set of norms and conventions as part of the contributor guidelines that will be included in our open source repos. My goal was to write a basic set of contributor guidelines that are generalized enough to be included in all of our repos, with room to add repo specific information where needed.

I opened this as a PR instead of pushing to `main` so that we could have a conversation around the contents of the document. 

- Is there anything that ought to be included or excluded?
- Are we in agreement that this is a sensible set of reqs?
- Is this generalized enough to include in all of our open source repos?

When building this document I referred to a few sources:
- https://opensource.guide/building-community/
- https://opensource.guide/starting-a-project/#writing-your-contributing-guidelines
- https://github.com/nayafia/contributing-template/blob/HEAD/CONTRIBUTING-template.md
- https://mozillascience.github.io/working-open-workshop/contributing/
- https://opensource.guide/best-practices/

### CODE REVIEW

**Review by chipcope — CHANGES_REQUESTED** [review]
@jtieri - So, I still don't know how to submit a PR...I know, disappointing...  So, if it works for you, I'll just drop some language below for inclusion?

> Before opening a PR, please review LICENSE.md and familiarize yourself with it's terms.  Please be advised that by opening a PR, you are granting Strangelove (or the owner of the relevant repository) a perpetual, worldwide, non-exclusive, no-charge, royalty-free, irrevocable license, in copyright and in patent, with respect to your Contribution and any portion thereof.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# Contributor Guidelines
+
+## Introduction 
+
+Welcome to the Strangelove contributor guidelines. We are excited to have you here and look forward to your contributions!
+Contributors are expected to adhere to the guidelines outlined in this document as well as our [code of conduct](./CODE_OF_CONDUCT.md).
+
+[Should Contributor Licensing information be included here?]
```
[the remark this answers is not in the document]
Comment (jtieri) [reply CONTRIBUTING.md:8]: I added @chipcope [proposed phrasing](https://github.com/strangelove-ventures/oss-repo-template-golang/pull/2#pullrequestreview-2091717448) to the end of the document under a new section titled `Contributor License Agreement`

**Comment by jtieri** [comment]
@beckettsean I have updated the document with a section on contributor license agreement and this should be ready for review again

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+##  Contributing
+
+There are many ways to contribute, from writing tutorials or blog posts, improving the documentation, 
+submitting bug reports and feature requests, or writing code which can be incorporated into the project itself.
+
+### Bug Reports
+
+First thing to note is that if you believe you have discovered a security vulnerability *DO NOT* use the issue tracker. 
+Please report it via email to security@strange.love. For more information on reporting security vulnerabilities,
```
Comment (beckettsean) [inline CONTRIBUTING.md:16]: We have a couple ways for folks to report. I do like having the more official automated way for people who are security aware, and this dead simple obvious way for others. 

Just calling it out that we might have two channels to monitor, but I'm okay with that.

**Review by beckettsean — APPROVED** [review]
LGTM!

File: CONTRIBUTING.md (hunk 2 above)
Comment (jtieri) [reply CONTRIBUTING.md:16]: good call out, this was before we recently decided to make use of the security vulnerability reporting via GitHub. I could add a sentence here pointing to our `SECURITY.md` file which would direct folks to the built in reporting functionality on GitHub but also leave our email in place as a secondary option?

Comment (jtieri) [reply CONTRIBUTING.md]: I ended up adding our email as a secondary option to `SECURITY.md` and pointed users to this document inside of `CONTRIBUTING.md`

**Review by beckettsean — APPROVED** [review]
Looks great!

**Review by beckettsean — APPROVED** [review]
trying again now that I'm part of the maintaining team

**Pull request merged**